### PR TITLE
Add option for graph refresh

### PR DESCRIPTION
**Pull request by agrare**
Adds graph refresh as an option for the foreman configuration and foreman providers

Related issue: https://github.com/ManageIQ/manageiq/issues/19888

### CODE REVIEW

File: app/models/manageiq/providers/foreman/inventory/parser/configuration_manager.rb [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    end
+  end
+
+  def configured_systems
+    collector.hosts.each do |host|
+      persister.configured_systems.build(
+        :manager_ref                        => host["id"].to_s,
+        :hostname                           => host["name"],
+        :last_checkin                       => host["last_compile"],
+        :build_state                        => host["buid"] ? "pending" : nil,
```
Comment (Fryguy) [inline app/models/manageiq/providers/foreman/inventory/parser/configuration_manager.rb]: Typo? `buid` -> `build`

File: app/models/manageiq/providers/foreman/inventory/collector/foreman.rb [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+      :architectures     => connection.all(:architectures),
+      :compute_profiles  => connection.all(:compute_profiles),
+      :domains           => connection.all(:domains),
+      :environments      => connection.all(:environments),
+      :realms            => connection.all(:realms),
+    }
+  end
+
+  # default taxonomy reference (locations and organizations)
+  def tax_refs
```
Comment (Fryguy) [inline app/models/manageiq/providers/foreman/inventory/collector/foreman.rb]: Can we just call it `taxonomy_refs`... I had to come down here to find out what this meant.

**Comment by Fryguy** [comment]
@agrare Is this ready to go?

File: app/models/manageiq/providers/foreman/inventory/collector/foreman.rb (hunk 2 above)
Comment (agrare) [reply app/models/manageiq/providers/foreman/inventory/collector/foreman.rb]: Sure, I just copied this from the old refresh parser but I can update the name

**Comment by agrare** [comment]
@Fryguy I have a little bit to go to get graph refresh equivalent to the old refresh_parser.  The ancestry rollups are the main one that I'm doing now.  I have the refresher_spec testing classic and graph so once graph passes the same specs I'll un-wip this.

File: app/models/manageiq/providers/foreman/configuration_manager/refresher.rb [hunk 3]
```diff
@@ -10,8 +10,12 @@ def parse_legacy_inventory(manager)
...
 
       def save_inventory(manager, target, hashes)
-        EmsRefresh.save_configuration_manager_inventory(manager, hashes, target)
-        EmsRefresh.queue_refresh(manager.provider.provisioning_manager) if hashes[:needs_provisioning_refresh]
+        if manager.inventory_object_refresh?
+          super
+        else
+          EmsRefresh.save_configuration_manager_inventory(manager, hashes, target)
+          EmsRefresh.queue_refresh(manager.provider.provisioning_manager) if hashes[:needs_provisioning_refresh]
+        end
```
Comment (agrare) [inline app/models/manageiq/providers/foreman/configuration_manager/refresher.rb]: Once we go to graph refresh as the only option we can remove the `save_inventory` override completely

**Comment by agrare** [comment]
@Fryguy okay got this passing the same tests as the old refresh

**Pull request merged**